src/components/core/geoChart/GeoChartOptionsWrapper.tsx

```diff
@@ -72,7 +88,7 @@ export class GeoChartOptionsWrapper extends BaseVisualization<IGeoChartInnerProp
             (item: IColorAssignment, legendIndex: number): IPushpinCategoryLegendItem => {
                 const name: string = isMappingHeaderAttributeItem(item.headerItem)
                     ? item.headerItem.attributeHeaderItem.name
-                    : EMPTY_SEGMENT_ITEM;
+                    : this.getEmptyHeaderString();
```

Comment:
We could have new class property to hold this static value and we don't have to calculate to same result for each loop of `colorAssignment.map`
```
private emptyHeaderString: string = this.getEmptyHeaderString();
```

Reply:
fixed

Reply:
Could you explain why we don't have?
```
private emptyHeaderString: string = this.getEmptyHeaderString();
```

Reply:
set value for `emptyHeaderString` at[here](https://github.com/gooddata/gooddata-react-components/blob/0316e55d0010a83cd686a41742299539b183030d/src/components/core/geoChart/GeoChartOptionsWrapper.tsx#L22). `this.props.intl` just exists in `renderVisualization`

Reply:
Sorry, I missed that.
So we could do as following
```
constructor(props) {
    super(props);
    this.emptyHeaderString = this.getEmptyHeaderString();
}
```
How do you think?

Reply:
fixed